src/kakarot/instructions/system_operations.cairo

```diff
@@ -555,42 +535,57 @@ namespace CallHelper {
     }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
         alloc_locals;
 
-        let is_reverted: felt = ExecutionContext.is_reverted(self=ctx);
-        local status: Uint256;
-        if (is_reverted != 0) {
-            // reverted contexts put a zero on the calling context's stack,
-            // so the calling context can handle failing cases
-            assert status = Uint256(low=0, high=0);
-        } else {
-            // otherwise, we put 1
-            assert status = Uint256(low=1, high=0);
-        }
+        // Pop ret_offset and ret_size
+        let (stack, popped) = Stack.pop_n(self=ctx.calling_context.stack, n=2);
+        let ret_offset = 2 ** 128 * popped[0].high + popped[0].low;
```

Comment:
thought i know the offset in memory can't probably be higher than a u64::max (eni calculated that the resulting memory expansion would cost billions), should we add some checks for overflow?

otherwise it could panic and revert the cairo vm?

Reply:
this does seem to be in the spec. If this brings an OOG issue, we should raise this in the corresponding component (probably the `Memory.store_n` where we would compute in advance the gas cost associated)